File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x<0){
            return false;
        }
        int y=x;
        int revnum=0;
        while(y!=0){
            int ld=y%10;
            revnum=(revnum*10)+ld;
            y=y/10;
        }
        if(revnum>INT_MAX || revnum<INT_MIN){
            return false;
        }
        if(revnum==x){
            return true;
        }
        return false;
    }
};

File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        y=x
        revnum=0
        while(y>0):
            ld=y%10
            revnum=(revnum*10)+ld
            y=y//10
        if revnum==x:
            return True
        else:
            return False

        